### PR TITLE
Change `push:` rule branch from `master` to `5.0.x` in workflow definitions

### PR DESCRIPTION
**Pull request by sasezaki**
Sorry. After #793, I noticed to have to change workflow branch to hook actions.

### CODE REVIEW

**Comment by sasezaki** [comment]
Notes:

On this PR, I would to change only workflow files.

There remains `master` on other markdown.
```
$ git ls-files | xargs grep master | grep -v composer.lock | grep -v CONTRIBUTING.md
README.md:* [Test suite](https://github.com/Roave/BetterReflection/blob/master/test/README.md)
docs/features.md:See [Loading a class from a string](https://github.com/Roave/BetterReflection/tree/master/docs/usage.md#Loading-a-class-from-a-string)
docs/features.md:See [types documentation](https://github.com/Roave/BetterReflection/tree/master/docs/types.md)
docs/features.md:See [AST extraction documentation](https://github.com/Roave/BetterReflection/tree/master/docs/ast-extraction.md)
Binary file test/unit/Fixture/autoload.phar matches
```

But, these can be redirected to main branch(5.0.x) by Github.

File: .github/workflows/benchmarks.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ on:
   pull_request:
   push:
     branches:
-      - "master"
+      - "5.0.x"
```
Comment (asgrim) [inline .github/workflows/benchmarks.yml:9]: I really wish there was a better way of doing this; the default branch changes every time there is a minor release. I found [this on Stack Overflow](https://stackoverflow.com/questions/64781462/github-actions-default-branch-variable) but I think it only goes in templates, not actual workflow files >.<

**Review by asgrim — APPROVED** [review]

**Comment by asgrim** [comment]
This is otherwise fine :+1:

**Comment by Ocramius** [comment]
My alternative for the future is to use https://github.com/laminas/laminas-ci-matrix-action/tree/71aa5bef47be3a9d3e1a3865f43caf16708ed847#usage

That would automate most of this stuff for us.

For now, this change is good as-is :)

**Review by Ocramius — APPROVED** [review]

**Comment by Ocramius** [comment]
Thanks @sasezaki!

**Pull request merged**